pkg/pubsub/adapter/converters/pubsub_pull.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"fmt"
+	cev2 "github.com/cloudevents/sdk-go/v2"
+	. "github.com/cloudevents/sdk-go/v2/event"
+	"github.com/google/knative-gcp/pkg/apis/events/v1beta1"
+	. "github.com/google/knative-gcp/pkg/pubsub/adapter/context"
+)
+
+func convertPubSubPull(ctx context.Context, msg *pubsub.Message) (*cev2.Event, error) {
```

Comment:
This is only used for pubsub channel, right?

Reply:
this is actually used for PullSubscriptions. I decided to leave the logic there. We had some E2E tests for it.
But we have our Pullsubscription resources as internal, so users should actually be using our four sources, and not this stuff....

As we don't do any opinionated conversions for the channel, the channel just uses:

`binding.ToEvent(ctx, cepubsub.NewMessage(msg))`